### ISSUE TITLE
AdaptiveScaffold.standardNavigationRail selectedIndex not nullable unlike normal NavigationRail

### ISSUE THREAD

**Issue by joshryandavis** [flutter/flutter#118888]
## Steps to Reproduce

1. Use  AdaptiveScaffold.standardNavigationRail and set selectedIndex to null.
2. Compare with using NavigationRail and setting selectedIndex to null. 

**Expected results:** selectedIndex can be set to null when no page in navigation is active.

**Actual results:** selectedIndex cannot be set to null. 

<details>
<summary>Code sample</summary>

```dart
```

</details>

<details>
  <summary>Logs</summary>

```
```

```
```

```
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @joshryandavis 
Can you provide `flutter doctor -v` and minimal code sample that shows the current behavior along with what should be the expected behavior ?

**Comment by joshryandavis** [comment]
> $ flutte

```
flutter doctor -v
> [!] Flutter (Channel beta, 3.7.0-1.4.pre, on macOS 13.1 22C65 darwin-arm64, locale en)
>     • Flutter version 3.7.0-1.4.pre on channel beta at /opt/homebrew/Caskroom/flutter/3.3.9/flutter
>     ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.18.7/libexec/bin/dart, which is not inside
>       your current Flutter SDK checkout at /opt/homebrew/Caskroom/flutter/3.3.9/flutter. Consider adding
>       /opt/homebrew/Caskroom/flutter/3.3.9/flutter/bin to the front of your path.
>     • Upstream repository https://github.com/flutter/flutter.git
>     • Framework revision 686fe913dc (13 days ago), 2023-01-10 16:14:01 -0800
>     • Engine revision ac5c83dd5f
>     • Dart version 2.19.0 (build 2.19.0-444.4.beta)
>     • DevTools version 2.20.0
>     • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
>       to perform update checks and upgrades.
> 
> [✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
>     • Android SDK at /Users/joshua/Library/Android/sdk
>     • Platform android-33, build-tools 33.0.0
>     • Java binary at: /Users/joshua/Library/Java/JavaVirtualMachines/openjdk-19.0.1-1/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build 19.0.1+10-21)
>     • All Android licenses accepted.
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 14.1)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Build 14B47b
>     • CocoaPods version 1.11.3
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [!] Android Studio
>     • Android Studio at /Applications/Android Studio Preview.app/Contents
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
>     ✗ Unable to find bundled Java version.
>     • Try updating or re-installing Android Studio.
> 
> [✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
>     • IntelliJ at /Users/joshua/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
> 
> [✓] IntelliJ IDEA Ultimate Edition (version 2022.3.2)
>     • IntelliJ at /Applications/IntelliJ IDEA 2022.3 EAP.app
>     • Flutter plugin version 71.3.6
>     • Dart plugin version 223.8617.8
> 
> [✓] VS Code (version 1.74.3)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension can be installed from:
>       🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
> 
> [✓] Connected device (2 available)
>     • macOS (desktop) • macos  • darwin-arm64   • macOS 13.1 22C65 darwin-arm64
>     • Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.87
>     ! Error: JD-iPhone is busy: Waiting to connect and unlock the device. Xcode will continue when JD-iPhone is
>       finished. (code -10)
> 
> [✓] HTTP Host Availability
>     • All required HTTP hosts are available
```

Examples:

```
import 'package:flutter/material.dart';
import 'package:flutter_adaptive_scaffold/flutter_adaptive_scaffold.dart';

class NavRailExample extends StatelessWidget {
  const NavRail({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const NavigationRail(
      leading: Text('Leading'),
      trailing: Text('Trailing'),
      selectedIndex: null, // can be null, so can be set to none selected
      onDestinationSelected: (int index) {},
      labelType: NavigationRailLabelType.selected,
      destinations: [
        NavigationRailDestination(
          icon: Icon(Icons.favorite_border),
          selectedIcon: Icon(Icons.favorite),
          label: Text('First'),
        ),
      ],
    );
  }
}

class NavRailAdaptiveExample extends StatelessWidget {
  const NavRailAdaptiveExample({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const AdaptiveScaffold.standardNavigationRail(
      leading: Text('Leading'),
      trailing: Text('Trailing'),
      selectedIndex: null, // can't be null, so can't be set to unselected.
      onDestinationSelected: (int index) {},
      labelType: NavigationRailLabelType.selected,
      destinations: [
        AdaptiveScaffoldDestination(
          icon: Icon(Icons.favorite_border),
          selectedIcon: Icon(Icons.favorite),
          label: Text('First'),
        ),
      ],
    );
  }
}

```

**Comment by darshankawar** [comment]
Thanks for the update. Seeing the same behavior as reported.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 3.3.10, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.3.10 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (15 hours ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.7.0-24.0.pre.87, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.7.0-24.0.pre.87 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c35efdaa68 (3 hours ago), 2023-01-24 02:57:00 +0100
    • Engine revision 7d3233d26d
    • Dart version 3.0.0 (build 3.0.0-154.0.dev)
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>
